input/script_utils.go

```diff
@@ -637,9 +637,21 @@ func SenderHTLCScriptTaproot(senderHtlcKey, receiverHtlcKey,
 	)
 }
 
+// maybeAppendSighashType appends a sighash type to the end of a signature if
+// the sighash type isn't sighash default.
+func maybeAppendSighash(sig Signature, sigHash txscript.SigHashType) []byte {
+	sigBytes := sig.Serialize()
+	if sigHash != txscript.SigHashDefault {
```

Comment:
Also think this should be used in the following functions:
- `ReceiverHTLCScriptTaprootRedeem` for `sweepSig` since this will likely be `SIGHASH_DEFAULT`. It doesn't need to be used for `senderSig` since it will be `SINGLE | ANYONECANPAY`.
- `ReceiverHTLCScriptTaprootTimeout` for `sweepSig` since it will likely be `SIGHASH_DEFAULT`.
- `TaprootHtlcSpendRevoke` since this sig here will likely be `SIGHASH_DEFAULT`
- `TaprootHtlcSpendSuccess` same as above